### PR TITLE
Add devcontainer with hls and unboxed tuple fix

### PR DESCRIPTION
**Pull request by hamishmack**
* adds a `skip` option to the plugin that can be used for `hls`.
* adds a `devcontainer` with `hls` and `plutus` packages.

To use the `skip` option add it to `cabal.project` or `cabal.project.local` file:

```
package example
  ghc-options: -fplugin-opt Language.PlutusTx.Plugin:skip

package plutus-use-cases
  ghc-options: -fplugin-opt Language.PlutusTx.Plugin:skip

package marlowe
  ghc-options: -fplugin-opt Language.PlutusTx.Plugin:skip

package marlowe-actus
  ghc-options: -fplugin-opt Language.PlutusTx.Plugin:skip
```

### CODE REVIEW

**Comment by michaelpj** [comment]
> uses a hie-bios with support for hie.project files
> uses hie.project and hie.project.local to turn on the skip options.

Hmm, I still think (cf https://github.com/input-output-hk/plutus/pull/2079#discussion_r442740260) that I'd prefer not to use a patched version and just suffer a bit of inconvenience for now (having to put the skip option in `cabal.project.local`).

File: nix/haskell-extra.nix [hunk 1]
```diff
@@ -58,31 +58,31 @@ in {
...
-          }."${location}"."${tag}";
+        sha256map = {
+          "https://github.com/bubba/brittany.git"."c59655f10d5ad295c2481537fc8abf0a297d9d1c" = "1rkk09f8750qykrmkqfqbh44dbx1p8aq1caznxxlw8zqfvx39cxl";
+        };
         inherit compiler-nix-name index-state checkMaterialization;
         # Invalidate and update if you change the version
-        plan-sha256 = "0a6c4lhnlm2lkic91ips0gb3hqlp3fk2aa01nsa8dhz9l8zg63da";
+        plan-sha256 = if pkgs.hostPlatform.isLinux
+          then "0jq1a2iyn394s3d2xag45d8ga32gn1i5bn5i63xd1jqllb85pcw3"
+          else "0l812savqj4xch9afky93zfkn5wvq6mz3q09v6mxkhcpnp6gshic";
```
Comment (michaelpj) [inline nix/haskell-extra.nix:79]: yikes

File: nix/sources.json [hunk 2]
```diff
@@ -1,4 +1,28 @@
 {
+    "cabal": {
```
Comment (michaelpj) [inline nix/sources.json:2]: unused?

File: example/.devcontainer/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM hamishmack/plutus-devcontainer
```
Comment (michaelpj) [inline example/.devcontainer/Dockerfile:1]: Does the user need all of these files as well as the image we make? That's a bit of a shame, can we package them up somehow?

File: nix/haskell-extra.nix (hunk 1 above)
Comment (michaelpj) [reply nix/haskell-extra.nix:79]: I think I might also have seen this... this seems very bad. `nix-tools` bug?

Comment (hamishmack) [reply nix/haskell-extra.nix]: Calculated plan is different as a result of `if os( darwin )` in cabal files.

File: nix/sources.json (hunk 2 above)
Comment (hamishmack) [reply nix/sources.json]: removed

File: example/.devcontainer/Dockerfile (hunk 3 above)
Comment (hamishmack) [reply example/.devcontainer/Dockerfile]: It works without the local Dockerfile now

**Comment by michaelpj** [comment]
We've got HLS and we've got the unboxed tuple fix.

**Pull request closed without merging**